examples/http_client.cr

```diff
@@ -0,0 +1,8 @@
+require "http"
+require "../src/db/pool"
+
+pool = DB::Pool.new { HTTP::Client.new(URI.parse("https://google.com")) }
```

Comment:
If we want `DB::Pool` to be used directly maybe the API for setting the pool options should be refactored a bit.

Otherwise it will be too fragile when new options are coming.